### PR TITLE
test: add type hints to the test_infra tests

### PR DESCRIPTION
**Pull request by tonyandrewmeyer**
Minor type hinting to make pyright happy:

```
$ env PYTHONPATH=.tox/static/lib/python3.10/site-packages/ .tox/static/bin/pyright test/test_infra.py 
WARNING: there is a new pyright version available (v1.1.317 -> v1.1.327).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

0 errors, 0 warnings, 0 informations 
```

Also renames an unused variable to `_` to get rid of an IDE warning.

Partial fix for #1007.

### CODE REVIEW

File: test/test_infra.py [hunk 1]
```diff
@@ -42,7 +43,7 @@ class InfrastructureTests(unittest.TestCase):
 
     def test_quote_backslashes(self):
         # ensure we're not using unneeded backslash to escape strings
-        issues = []
+        issues : typing.List[typing.Tuple[str, int, str]] = []
```
Comment (benhoyt) [inline test/test_infra.py:46]: I thought it was odd that Flake8 didn't complain about the [PEP8 pet peeve](https://www.python.org/dev/peps/pep-0008/#pet-peeves) of whitespace before a colon, but I see we have `E203` ignored in our `pyproject.toml`. Odd -- not sure why. Can we please un-ignore that and fix these? From a quick check, the only instances are the ones in this file. Needs to be done in the tool.autopep8 and tool.flake8 section.

**Review by benhoyt — CHANGES_REQUESTED** [review]
Thanks! I think we should also add "test/test_infra.py" to the `pyproject.toml` `tool.pyright.include` list to keep that up to date and checking the files we've done.

File: test/test_infra.py (hunk 1 above)
Comment (tonyandrewmeyer) [reply test/test_infra.py:46]: > I thought it was odd that Flake8 didn't complain about the [PEP8 pet peeve](https://www.python.org/dev/peps/pep-0008/#pet-peeves) of whitespace before a colon, but I see we have `E203` ignored in our `pyproject.toml`. Odd -- not sure why. Can we please un-ignore that and fix these? From a quick check, the only instances are the ones in this file. Needs to be done in the tool.autopep8 and tool.flake8 section.

Good, I don't like that either and was just relying on the tooling :)

**Review by benhoyt — APPROVED** [review]
Excellent, thanks!

**Pull request merged**